device/core/src/module_client.ts

```diff
@@ -182,28 +184,56 @@ export class ModuleClient extends InternalClient {
     this._methodClient.invokeMethod(deviceId, actualModuleId, actualMethodParams as MethodParams, actualCallback);
   }
 
+  /**
+   * Invokes a method on a downstream device or on another module on the same Edge device. Please note that this feature only works when
+   * the module is being run as part of an Edge device.
+   *
+   * @param deviceId      target device identifier
+   * @param moduleId      target module identifier on the device identified with the `deviceId` argument
+   * @param methodParams  parameters of the direct method call
+   * @param [callback]    optional callback that will be invoked either with an Error object or the result of the method call.
+   * @returns {Promise<MethodResult> | void} Promise if no callback function was passed, void otherwise.
+   */
+  invokeMethod(deviceId: string, methodParams: MethodParams, callback: Callback<MethodResult>): void;
+  invokeMethod(deviceId: string, moduleId: string, methodParams: MethodParams, callback: Callback<MethodResult>): void;
+  invokeMethod(deviceId: string, moduleIdOrMethodParams: string | MethodParams, methodParamsOrCallback: MethodParams | Callback<MethodResult>, callback?: Callback<MethodResult>): Promise<MethodResult> | void {
+    if (callback) {
+      return this._invokeMethod(deviceId, moduleIdOrMethodParams as string, methodParamsOrCallback as MethodParams, callback);
+    } else if ((methodParamsOrCallback instanceof Function)) {
```

Comment:
Unnecessary set of parentheses, right?

Reply:
Done.

---
In reply to: [218892032](https://github.com/Azure/azure-iot-sdk-node/pull/372#discussion_r218892032) [](ancestors = 218892032)